qiskit/dagcircuit/_dagcircuit.py

```diff
@@ -313,15 +331,18 @@ def apply_operation_back(self, op, qargs=None, cargs=None, condition=None):
         al = [qargs, all_cbits]
         for q in itertools.chain(*al):
             ie = list(self.multi_graph.predecessors(self.output_map[q]))
+
             if len(ie) != 1:
                 raise DAGCircuitError("output node has multiple in-edges")
 
-            self.multi_graph.add_edge(ie[0], self._max_node_id,
+            self.multi_graph.add_edge(ie[0], self.id_to_node[self._max_node_id],
                                       name="%s[%s]" % (q[0].name, q[1]), wire=q)
             self.multi_graph.remove_edge(ie[0], self.output_map[q])
-            self.multi_graph.add_edge(self._max_node_id, self.output_map[q],
-                                      name="%s[%s]" % (q[0].name, q[1]), wire=q)
-
+            self.multi_graph.add_edge(self.id_to_node[self._max_node_id], self.output_map[q],
```

Comment:
I think the first statement adds an edge from the predecessor of `output_map[q]` to `_max_node_id`. There is no change to the logic here from master it has only been updated so that it uses `DAGNodes`. I can rename the variables to make what is going on a bit clearer though!